### PR TITLE
Add ethereum regression tests

### PR DESCRIPTION
**Pull request by mossberg**
Add tests for various simple contracts 

This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/763)

### CODE REVIEW

**Comment by yan** [comment]
Reviewed 8 of 9 files at r1.
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[tests/test_binaries.py, line 156 at r1](https://reviewable.io:443/reviews/trailofbits/manticore/763#-L5zBStOjHq2YVdT1Od9:-L5zBStOjHq2YVdT1OdA:biqe5gr) ([raw file](https://github.com/trailofbits/manticore/blob/aae8f8eb0a64b3798db984f2ee9e5db40b448cec/tests/test_binaries.py#L156)):*
> ```Python
>     def test_eth_760(self):
>         self._simple_cli_run('760.sol')
> ```

A lot of these names/paths are repeated. If we misname a test, it won't execute. Why not?:

```python
# test_eth_705 as is due to added set up/teardown

def test_eth_regressions(self):
    contracts = [676, 678, 701, 714, 735, 760]
    for contract in contracts:
        self._simple_cli_run('{}.sol'.format(contract))
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/trailofbits/manticore/763)*

**Comment by mossberg** [comment]
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[tests/test_binaries.py, line 156 at r1](https://reviewable.io:443/reviews/trailofbits/manticore/763#-L5zBStOjHq2YVdT1Od9:-L5zC_iO1llwhj8HOBBu:bhnb3rh) ([raw file](https://github.com/trailofbits/manticore/blob/aae8f8eb0a64b3798db984f2ee9e5db40b448cec/tests/test_binaries.py#L156)):*
<details><summary><i>Previously, yan (Yan Ivnitskiy) wrote…</i></summary><blockquote>

A lot of these names/paths are repeated. If we misname a test, it won't execute. Why not?:

```python
# test_eth_705 as is due to added set up/teardown

def test_eth_regressions(self):
    contracts = [676, 678, 701, 714, 735, 760]
    for contract in contracts:
        self._simple_cli_run('{}.sol'.format(contract))
```
</blockquote></details>

Good point. I originally did it this way so when a test fails, we'll immediately know specifically which one. But i do like that better, will make the change

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/trailofbits/manticore/763)*

**Comment by yan** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/trailofbits/manticore/763#-:-L5zDpv2r6Ttx1f7h7YA:bnfp4nl)*

File: tests/test_binaries.py [hunk 1]
```diff
@@ -88,7 +97,6 @@ def testArgumentsAssertions(self):
         filename = os.path.abspath(os.path.join(dirname, 'binaries/arguments_linux_amd64'))
         self.assertTrue(filename.startswith(os.getcwd()))
         filename = filename[len(os.getcwd())+1:]
-        data = file(filename,'rb').read()
         workspace = '%s/workspace'%self.test_dir
         assertions = '%s/assertions.txt'%self.test_dir
         file(assertions,'w').write('0x0000000000401003 ZF == 1')
```
Comment (disconnect3d) [inline tests/test_binaries.py]: `file` -> `open`

**Pull request merged**